bip-0054.md

```diff
@@ -0,0 +1,231 @@
+```
+  BIP: 54
+  Layer: Consensus (soft fork)
+  Title: Consensus Cleanup
+  Author: Antoine Poinsot <mail@antoinep.com>
+          Matt Corallo <bips@bluematt.me>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-0054
+  Status: Draft
+  Type: Standards Track
+  Created: 2024-04-11
+  License: CC0-1.0
+```
+
+## Abstract
+
+This document proposes new consensus rules in order to fix the timewarp attack, reduce the worst
+case block validation time, prevent Merkle tree weaknesses, and avoid duplicate transactions without
+[bip-0030][BIP30] validation.
+
+## Motivation
+
+This proposal addresses a number of long-standing vulnerabilities and weaknesses in the Bitcoin
+protocol. Bundling these fixes together allows to overcome the fixed cost of deploying a Bitcoin
+soft fork.
+
+The timewarp bug permits a majority hashrate attacker to arbitrarily increase the block rate,
+allowing them to steal block subsidy from future miners and increase validation costs to nodes that
+have to deal with the increased average transaction rate. By strategically setting the block
+timestamp, the timewarp bug lets miners bring down the difficulty to its minimum within 38 days of
+starting the attack. The existence of this bug not only significantly empowers a 51% attacker, but
+also makes it notably harder to reason about miners' incentives. It could indeed be in the interest
+of short-sighted miners as well as short-sighted users to exploit this vulnerability in a small
+enough proportion to increase the block rate without fatally hurting the network, as the effectively
+increased block space would — all other things equal — bring fee rates down for users.
+
+Specially crafted blocks may be expensive to process, with validation times ranging from a few
```

Comment:
Suggest separating the different issues in the motivation section with individual headers to improve clarity. Better yet, a focused BIP per issue and its mitigation.